wallets/react/src/hub/autoConnect.ts

```diff
@@ -60,21 +65,49 @@ async function eagerConnect(
     targetNamespaces.push([namespaceInput, result]);
   });
 
-  const finalResult = targetNamespaces.map(([info, namespace]) => {
-    const evmChain = legacyIsEvmNamespace(info)
-      ? convertNamespaceNetworkToEvmChainId(info, allBlockChains || [])
-      : undefined;
-    const chain = evmChain || info.network;
+  const connectNamespacesPromises = targetNamespaces.map(
+    ([info, namespace]) => {
+      const evmChain = legacyIsEvmNamespace(info)
+        ? convertNamespaceNetworkToEvmChainId(info, allBlockChains || [])
+        : undefined;
+      const chain = evmChain || info.network;
 
-    return async () => await namespace.connect(chain);
-  });
+      return async () => await namespace.connect(chain);
+    }
+  );
 
   /**
    * Sometimes calling methods on a instance in parallel, would cause an error in wallet.
    * We are running a method at a time to make sure we are covering this.
    * e.g. when we are trying to eagerConnect evm and solana on phantom at the same time, the last namespace throw an error.
    */
-  return await sequentiallyRun(finalResult);
+  const connectNamespacesResult = await sequentiallyRun(
+    connectNamespacesPromises
+  );
+
+  const failedNamespaces: LegacyNamespaceInputForConnect[] = [];
+  const successNamespaces: LegacyNamespaceInputForConnect[] = [];
+  connectNamespacesResult.forEach((result, index) => {
+    if (result instanceof Error) {
+      failedNamespaces.push(targetNamespaces[index][0]);
+    } else {
+      successNamespaces.push(targetNamespaces[index][0]);
+    }
+  });
+
+  if (failedNamespaces.length > 0) {
+    lastConnectedWalletsFromStorage.removeNamespacesFromWallet(
+      type,
+      failedNamespaces.map((namespace) => namespace.namespace)
+    );
+  }
+
+  const atLeastOneNamespaceConnectedSuccessfully = successNamespaces.length > 0;
```

Comment:
you can remove `successNamespaces` and consoder it as:
```suggestion
  const atLeastOneNamespaceConnectedSuccessfully = connectNamespacesResult.length - failedNamespaces.length > 0;
```

I'm suggesting this since you only need the length, and not the input itselt.

Reply:
Good point, applied